sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -184,15 +189,62 @@ abstract class JdbcDialect extends Serializable {
   /**
    * Rename an existing table.
    *
-   * TODO (SPARK-32382): Override this method in the dialects that don't support such syntax.
-   *
    * @param oldTable The existing table.
    * @param newTable New name of the table.
    * @return The SQL statement to use for renaming the table.
    */
   def renameTable(oldTable: String, newTable: String): String = {
     s"ALTER TABLE $oldTable RENAME TO $newTable"
   }
+
+  /**
+   * Alter an existing table.
+   * TODO (SPARK-32523): Override this method in the dialects that have different syntax.
+   *
+   * @param tableName The name of the table to be altered.
+   * @param changes Changes to apply to the table.
+   * @return The SQL statements to use for altering the table.
+   */
+  def alterTable(tableName: String, changes: Seq[TableChange]): Array[String] = {
+    val updateClause = ArrayBuilder.make[String]
+    for (change <- changes) {
+      change match {
+        case add: AddColumn =>
+          add.fieldNames match {
+            case Array(name) =>
+              val dataType = JdbcUtils.getJdbcType(add.dataType(), this).databaseTypeDefinition
+              updateClause += s"ALTER TABLE $tableName ADD COLUMN $name $dataType"
+          }
+        case rename: RenameColumn =>
+          rename.fieldNames match {
+            case Array(name) =>
+              updateClause += s"ALTER TABLE $tableName RENAME COLUMN $name TO ${rename.newName}"
+          }
+        case delete: DeleteColumn =>
+          delete.fieldNames match {
+            case Array(name) =>
+              updateClause += s"ALTER TABLE $tableName DROP COLUMN $name"
+          }
+        case update: UpdateColumnType =>
+          update.fieldNames match {
+            case Array(name) =>
+              val dataType = JdbcUtils.getJdbcType(update.newDataType(), this)
+                .databaseTypeDefinition
+              updateClause += s"ALTER TABLE $tableName ALTER COLUMN $name $dataType"
+          }
+        case update: UpdateColumnNullability =>
+          update.fieldNames match {
+            case Array(name) =>
+              val nullable = if (update.nullable()) "NULL" else "NOT NULL"
+              updateClause += s"ALTER TABLE $tableName ALTER COLUMN $name SET " + nullable
+          }
+        case _ =>
+          throw new SQLFeatureNotSupportedException(s"${this.getClass.getName}.alterTable" +
```

Comment:
@MaxGekk I changed to ```SQLFeatureNotSupportedException```. I think it might be better than ```NotImplementedError``` because it doesn't need scalastyle:off

Reply:
ditto, `this.getClass.getName` is not useful here. Just say `Unsupported TableChange: $change`